api/include/opentelemetry/context/context.h

```diff
@@ -92,22 +90,21 @@ class Context
 
 private:
   // A linked list to contain the keys and values of this context node
-  class DataList
+  struct DataList
   {
-  public:
-    char *key_;
+    char *key_ = nullptr;
 
-    nostd::shared_ptr<DataList> next_;
+    size_t key_length_ = 0UL;
 
-    size_t key_length_;
+    nostd::shared_ptr<DataList> next_;
```

Comment:
Why reordering the members here ?

Reply:
If I recall, it started complaining about member ordering when I added the copy ctor that it asked [here](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/11902741932/job/33168372488?pr=3151#step:4:55)

I just re-ran the the test with `next_` back up above `key_length_` and cppcheck seems to be more accepting of it, so it must've been something else while the changes were still work in progress on my end.

Do you want me to revert this ordering?

Reply:
Yes, please revert. Better not take chances with binary layout if we can avoid it.

In theory, theory and practice are the same.
In practice, they are different.

Reply:
Thinking more about it, I think changing the order of members is a breaking change:

* an instrumented library A built with otel-cpp v1.17.0 (without this change)
* an instrumented library B built with otel-cpp v1.18.0 (with this change)

should see the same bits in the same place, when manipulating a runtime context using their own copy of `class Context`, which includes `DataList`, when both libraries A and B are linked in the same application.

The runtime context is shared between all instrumented libraries, because it comes from a global singleton.

Same for the baggage propagator.

Reply:
I remember now; the issue came up not with cppcheck but with `-Werror=reorder` in maintainer mode CI checks and it felt more natural to initialize the key and key length back-to-back so I had swapped the order of the member variables instead the order of init in the copy ctor.

In any case, they are reverted to their original order and the change is in now in the copy constructor to respect the original order of initialization.